### PR TITLE
[Feature] add mmdeploy for human perception

### PR DESCRIPTION
**Pull request by KHao123**
add mmdeploy for human perception (mmdet and mmpose)

- add mmdeploy convertion scripts and configs 

- add estimator to adapt mmdeploy

### CODE REVIEW

File: docs/en/tools/mmdeploy.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Tool mmdeply
+
+- [Overview](#overview)
+- [Installation](#Installation)
+- [Run](#Run)
+
+### Overview
+This tool coverts human perception module into TensorRT with mmdeploy.
```
Comment (LazyBusyYang) [inline docs/en/tools/mmdeploy.md:8]:
```suggestion
This tool converts human perception pytorch module into TensorRT engine with mmdeploy.
```

File: scripts/run_mmdeply.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+mkdir ./weight/mmdet_faster_rcnn/
```
Comment (LazyBusyYang) [inline scripts/run_mmdeply.sh:1]: What about split it into 2 scripts, `deploy_mmdet_faster_rcnn` and `deploy_mmpose_hrnet`.

File: tools/deploy.py [hunk 3]
```diff
@@ -0,0 +1,490 @@
+# Copyright (c) OpenMMLab. All rights reserved.
```
Comment (LazyBusyYang) [inline tools/deploy.py:1]: Is it all same as the file from mmdeploy official repo? If so, we'd better not to commit it into our repo, just advise users to clone one, and make mmdeploy path as an argument. We haven't copy files except configs from mmdet/mmpose before.

File: xrmocap/human_perception/bbox_detection/mmdet_trt_detector.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    import traceback
+    stack_str = ''
+    for line in traceback.format_stack():
+        if 'frozen' not in line:
+            stack_str += line + '\n'
+    import_exception = traceback.format_exc() + '\n'
+    import_exception = stack_str + import_exception
+
+
+class MMdettrtDetector:
```
Comment (LazyBusyYang) [inline xrmocap/human_perception/bbox_detection/mmdet_trt_detector.py:27]:
```suggestion
class MMdetTrtDetector:
```
or
```suggestion
class MMdeployTrtDetector:
```
I prefer the second.

File: xrmocap/human_perception/bbox_detection/mmdet_trt_detector.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        """
+        image_array = video_to_array(input_path=video_path, logger=self.logger)
+        ret_list = self.infer_array(
+            image_array=image_array,
+            disable_tqdm=disable_tqdm,
+            multi_person=multi_person)
+        return ret_list
+
+
+def process_mmdet_results(mmdet_results: list,
```
Comment (LazyBusyYang) [inline xrmocap/human_perception/bbox_detection/mmdet_trt_detector.py:203]: What about `from . mmdet_detector import process_mmdet_results`?

File: xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+    import traceback
+    stack_str = ''
+    for line in traceback.format_stack():
+        if 'frozen' not in line:
+            stack_str += line + '\n'
+    import_exception = traceback.format_exc() + '\n'
+    import_exception = stack_str + import_exception
+
+
+class MMposetrtTopDownEstimator:
```
Comment (LazyBusyYang) [inline xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py:36]:
```suggestion
class MMdeployTrtTopDownEstimator:
```

File: xrmocap/human_perception/bbox_detection/mmdet_trt_detector.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+                 deploy_cfg: str,
+                 model_cfg: str,
+                 backend_files: str,
+                 device: str = 'cuda',
+                 batch_size: int = 1,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """Init a detector from mmdetection.
+
+        Args:
+            mmdet_kwargs (dict):
```
Comment (LazyBusyYang) [inline xrmocap/human_perception/bbox_detection/mmdet_trt_detector.py:40]: Docstring not updated.

File: xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py [hunk 8]
```diff
@@ -0,0 +1,555 @@
...
+                 deploy_cfg: str,
+                 model_cfg: str,
+                 backend_files: str,
+                 device: str = 'cuda',
+                 bbox_thr: float = 0.0,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """Init a detector from mmpose.
+
+        Args:
+            mmpose_kwargs (dict):
```
Comment (LazyBusyYang) [inline xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py:48]: Docstring not updated.

File: xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py [hunk 9]
```diff
@@ -0,0 +1,555 @@
...
+            keypoints2d = Keypoints(
+                kps=kps2d_arr,
+                mask=mask_arr,
+                convention=self.get_keypoints_convention_name(),
+                logger=self.logger)
+        else:
+            keypoints2d = None
+        return keypoints2d
+
+    def inference_top_down_pose_model(self,
```
Comment (LazyBusyYang) [inline xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py:348]: Is there a chance import this function from mmdeploy?

File: xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py (hunk 6 above)
Comment (LazyBusyYang) [inline xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py:36]: What about creating a sub-class of `MMposeTopDownEstimator`?

File: tools/deploy.py (hunk 3 above)
Comment (KHao123) [reply tools/deploy.py:1]: Mostly the same as official repo. I change the import manner to adpat isort pipeline.

File: xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py (hunk 9 above)
Comment (KHao123) [reply xrmocap/human_perception/keypoints_estimation/mmpose_trt_top_down_estimator.py:348]: this function is rewrited to adapt the input of the original function `infer_array` of `mmpose_top_down_estimator`

**Review by LazyBusyYang — APPROVED** [review]

**Pull request merged**